playwright/_impl/_connection.py

```diff
@@ -83,6 +84,7 @@ async def inner_send(
         if self._connection._error:
             error = self._connection._error
             self._connection._error = None
+            callback.future.cancel()
```

Comment:
Perhaps we should first check the error, and if there is no error, only then send the message?